File: hw/chalan/u2/hw25ConnectiontotheDB/JAVA/ContacksBook/src/main/java/ec/edu/espe/contactsbook/utils/MongoConnection.java
package ec.edu.espe.contactsbook.utils;

import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;

public class MongoConnection {
    private static MongoClient client;
    private static MongoDatabase database;

    public static MongoDatabase connect() {
        if (database == null) {
            try {
                String uri = "mongodb+srv://kevin:kevin2001@cluster0.oxinj5p.mongodb.net/";
                client = MongoClients.create(uri);
                database = client.getDatabase("ContactsDB");

                System.out.println("✅ Conectado a MongoDB!");
            } catch (Exception e) {
                System.out.println("❌ Error al conectar a MongoDB:");
                e.printStackTrace();
            }
        }
        return database;
    }
}

File: hw/chalan/u2/hw25ConnectiontotheDB/contact_appPython/database/mongo_connection.py
from pymongo import MongoClient

class MongoConnection:
    def __init__(self, 
                 uri="mongodb+srv://kevin:kevin2001@cluster0.oxinj5p.mongodb.net/",
                 db_name="contac"):
        
        self.client = MongoClient(uri)
        self.db = self.client[db_name]

    def get_collection(self, name):
        return self.db[name]
